content/en/docs/concepts/extend-kubernetes/compute-storage-net/resource-drivers.md

```diff
@@ -0,0 +1,286 @@
+    // can be allocated for it (for example, two GPUs requested but
+    // the node only has one).
+    //
+    // The result of the check is in ClaimAllocation.UnsuitableNodes.
+    // An error indicates that the entire check must be repeated.
+    UnsuitableNodes(ctx context.Context, pod *v1.Pod, claims []*ClaimAllocation, potentialNodes []string) error
+```
+
+
+## Kubelet-plugin implementation
+
+Resource driver's kubelet-plugin's main purpose is to ensure the ResourceClaims, that the Pod will
+be using on Node, have all resources ready for Pod to take in use.
+
+```gRPC
+    // NodePrepareResources prepares several ResourceClaims
+    // for use on the node. If an error is returned, the
+    // response is ignored. Failures for individual claims
+    // can be reported inside NodePrepareResourcesResponse.
+    NodePrepareResources(context.Context, *NodePrepareResourcesRequest) (*NodePrepareResourcesResponse, error)
+    // NodeUnprepareResources is the opposite of NodePrepareResources.
+    // The same error handling rules apply,
+    NodeUnprepareResources(context.Context, *NodeUnprepareResourcesRequest) (*NodeUnprepareResourcesResponse, error)
+```
+
+### Example {#example-pod}
+
+Suppose a Kubernetes cluster is running a resource driver gpu.resource.example.com with Resource
+Class `example.example.com`. Here is an example of a pod requesting this resource to run a demo
+workload:
+
+```yaml
+apiVersion: gpu.resource.example.com/v1alpha1
+kind: GpuClaimParameters
+metadata:
+  name: single-gpu
+spec:
+  count: 1
+---
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClaim
+metadata:
+  name: gpu-test
+spec:
+  resourceClassName: gpu.example.com
+  parametersRef:
+    apiGroup: gpu.resource.example.com/v1alpha1
+    kind: GpuClaimParameters
+    name: single-gpu
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: gpu-test4
+  name: pod0
+  labels:
+    app: pod
+spec:
+  containers:
+  - name: container1
+    image: ubuntu:22.04
+    command: ["bash", "-c"]
+    args: ["export; sleep 9999"]
+    resources:
+      claims:
+      - name: gpus
+  resourceClaims:
+  - name: gpus
+    source:
+      resourceClaimTemplateName: gpu-test
+# This Pod wants to use ResourceClaim gpu-test that needs 1 device of ResourceClass
+# gpu.example.com, that is handled by the Resource Driver gpu.resource.example.com .
+#
+# The Resource Driver will allocate the needed resource to ResourceClaim and ensure it is
+# ready to use, only then the Pod will start.
+```
+
+## Resource Driver deployment
+
+The recommended way to deploy a resource driver is a Deployment for controller part and a DaemonSet
+for kubelet-plugin part. It is also possible to deploy it as a package for your node's
+operating system, or manually.
+
+Kubelet will be using gRPC interface to interact with Resource Driver's kubelet-plugin. No special
+permissions are required for resource-driver.
+
+ResourceClass should be part of Resource Driver deployment.
```

Comment:
```suggestion
When you deploy a resource driver, you typically also define at least one ResourceClass using that driver.
```

Reply:
Doesn't this seem more like guidance for additional post-installation steps rather than suggestion to include pre-defined ResourceClass into resource driver installation package?

Reply:
```suggestion
When creating a resource driver you should include at least one ResourceClass for it in the resource driver deployment package.
```